sbpy/data/ephem.py

```diff
@@ -127,35 +129,151 @@ def from_horizons(cls, targetids, id_type='smallbody',
+        >>> eph = Ephem.from_mpc('2P', epochs=epochs, location='568')  # doctest: +REMOTE_DATA +IGNORE_OUTPUT
 
-        not yet implemented
+
+        Notes
+        -----
+        See `~astroquery.mpc.MPC.get_ephemerides` and the Minor Planet
+        Ephemeris Service user's guide [MPES]_ for details, including
+        accetable target names.
+
+
+        References
+        ----------
+        .. [MPES] Wiliams, G. The Minor Planet Ephemeris Service.
+           https://minorplanetcenter.org/iau/info/MPES.pdf
+
+        .. [OBSCODES] IAU Minor Planet Center.  List of observatory
+           codes. https://minorplanetcenter.org/iau/lists/ObsCodesF.html
 
         """
 
+        # parameter check
+        if isinstance(epochs, dict):
+            start = epochs['start']  # required
+            step = epochs.get('step')
+            stop = epochs.get('stop')
+            number = epochs.get('number')
+
+            if isinstance(start, (float, int)):
+                start = Time(start, format='jd', scale='utc')
+
+            if isinstance(stop, (float, int)):
+                stop = Time(stop, format='jd', scale='utc')
+
+            if step is not None:
+                step = u.Quantity(step)
+                if step.unit not in (u.d, u.h, u.m, u.s):
+                    raise ValueError(
+                        'step must have units of days, hours, minutes,'
+                        ' or seconds')
+
+            if stop is not None:
+                if step is None:
+                    raise ValueError(
+                        'step is required when start and stop are provided')
+
+                # start and stop both defined, estimate number of steps
+                dt = (Time(stop).jd - Time(start).jd) * u.d
+                number = int((dt / step).decompose()) + 1
+        else:
+            start = None
+
+            if epochs is None:
+                epochs = Time.now()
+
+            if not np.iterable(epochs) or isinstance(epochs, str):
+                epochs = [epochs]
+
+            # check for Julian dates
+            for i in range(len(epochs)):
+                if isinstance(epochs[i], (float, int)):
+                    epochs[i] = Time(epochs[i], format='jd', scale='utc')
+
+        # get ephemeris
+        if start is None:
+            eph = []
+            for i in range(len(epochs)):
+                start = Time(epochs[i], scale='utc')
+                e = MPC.get_ephemeris(targetid, location=location,
+                                      start=start, number=1, **kwargs)
+                e['Date'] = e['Date'].iso  # for vstack to work
+                eph.append(e)
+            eph = vstack(eph)
+            eph['Date'] = Time(eph['Date'], scale='utc')
+        else:
+            eph = MPC.get_ephemeris(targetid, location=location,
+                                    start=start, step=step, number=number,
+                                    **kwargs)
+
```

Comment:
one more thing: could you add a column to your resulting table named `timescale` that describes the time scale of the epoch in your table (e.g., `UTC` for UT). This is necessary for compatibility. Thanks!

Reply:
Sure.  Sounds like a check to add to Ephem.__init__?